### PR TITLE
last feature PR before 1.1

### PR DESCRIPTION
**Pull request by franz**
kernel compilation speedup + various fixes

### CODE REVIEW

File: lib/CL/devices/pthread/pthread.c [hunk 1]
```diff
@@ -476,7 +476,7 @@ void pocl_pthread_update_event (cl_device_id device, cl_event event, cl_int stat
       if (event->queue->properties & CL_QUEUE_PROFILING_ENABLE)
         event->time_end = device->ops->get_timer_value(device->data);
 
-
+      POCL_DEBUG_EVENT_TIME (event, "NDRange Kernel        ");
```
Comment (pjaaskel) [inline lib/CL/devices/pthread/pthread.c]: Why the white space? Looks weird here.

Comment (franz) [reply lib/CL/devices/pthread/pthread.c]: string-justify... all timing info messages have the same length, originally intended for nicer output when run with POCL_DEBUG=timing

**Comment by jrprice** [comment]
FYI this PR crashes with a multi-device CG benchmark that I use to test the async stuff in the CUDA backend - will try and investigate.

**Comment by franz** [comment]
@jrprice FWIW the only changes directly related to CUDA are the module verification (i enabled it by default) or the event-wait. Wildly guessing it's the module verification

File: CMakeLists.txt [hunk 2]
```diff
@@ -387,18 +387,63 @@ endmacro()
 
 if(UNIX)
   include(CheckCSourceCompiles)
+
+  # don't allow implicit function declarations
+  set(CMAKE_REQUIRED_FLAGS "-std=c99")
```
Comment (jrprice) [inline CMakeLists.txt]: This causes the `HAVE_CLOCK_GETTIME` check to fail on my system (with various versions of GCC).

Using `-std=gnu99` instead or adding `-D_POSIX_C_SOURCE=199309L` fixes it.

Comment (franz) [reply CMakeLists.txt]: There should be a ``#define _DEFAULT_SOURCE`` in the HAVE_CLOCK_GETTIME test with commit ff42b03 - according to gc manual for _DEFAULT_SOURCE, "the effect is to enable features from the 2008 edition of POSIX". Weird thing is, i can't see that commit (and a bunch of others) in this PR.

Comment (jrprice) [reply CMakeLists.txt]: `_DEFAULT_SOURCE` is there, but it was only introduced in glibc 2.19, and my glibc is a couple of revs older than that (using RHEL 7.3).

Comment (franz) [reply CMakeLists.txt]: @jrprice OK thanks, i forgot that _DEFAULT_SOURCE is quite recent.

File: lib/CL/devices/pthread/pthread.c (hunk 1 above)
Comment (pjaaskel) [reply lib/CL/devices/pthread/pthread.c]: Isn't this better be done at the callee side of this macro? No biggie, just looks a bit funny.

**Comment by franz** [comment]
dropped the patch causing problems for CUDA, plus slightly tuned the portability check macros

**Comment by franz** [comment]
After spending some time with ThreadSanitizer + CUDA, i've decided to postpone all CUDA patches (except module verification) until after pocl 1.1 release. It'll require more work than makes sense to do before release.

File: CMakeLists.txt (hunk 2 above)
Comment (Oblomov) [reply CMakeLists.txt]: Setting `_POSIX_C_SOURCE` to 200809L should work too.

**Pull request merged**